x/pairing/keeper/msg_server_relay_payment.go

```diff
@@ -392,3 +391,102 @@ func appendRelayPaymentDetailsToEvent(from map[string]string, uniqueIdentifier u
 	}
 	return to
 }
+
+// distributeRewards is the main function for reward distribution for providers and delegators
+func (k Keeper) distributeRewards(ctx sdk.Context, providerAddr sdk.AccAddress, chainID string, block uint64, totalReward math.Int) error {
+	epoch, _, err := k.epochStorageKeeper.GetEpochStartForBlock(ctx, block)
```

Comment:
why not use `block` directly?

Reply:
because when I fetch stakeStorage, the function constructs a key that is dependent on the block value. Since stakeStorage objects are saved with keys that always contain blocks of epoch starting points, I need to get the start of the epoch